From 166213be08cbf171478f646c14d1f3035e83046e Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:21 +0100
Subject: [PATCH 50/52] Revert "Let autoload act on fwd declares."

This reverts commit aaa35c7ff707ee66f9511419af65a7809bda7366.
---
 interpreter/llvm-project/clang/lib/Sema/SemaLookup.cpp | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/interpreter/llvm-project/clang/lib/Sema/SemaLookup.cpp b/interpreter/llvm-project/clang/lib/Sema/SemaLookup.cpp
index b945a530c1..02b1a045df 100644
--- a/interpreter/llvm-project/clang/lib/Sema/SemaLookup.cpp
+++ b/interpreter/llvm-project/clang/lib/Sema/SemaLookup.cpp
@@ -2273,14 +2273,8 @@ bool Sema::LookupName(LookupResult &R, Scope *S, bool AllowBuiltinCreation,
       }
   } else {
     // Perform C++ unqualified name lookup.
-    if (CppLookupName(R, S)) {
-      if (R.isSingleResult())
-        if (const TagDecl *TD = dyn_cast<TagDecl>(R.getFoundDecl())) {
-          if (!TD->getDefinition() && ExternalSource)
-            ExternalSource->LookupUnqualified(R, S);
-        }
+    if (CppLookupName(R, S))
       return true;
-    }
   }
 
   // If we didn't find a use of this identifier, and if the identifier
-- 
2.43.5

